prov/shm/src/smr_av.c

```diff
@@ -70,6 +70,12 @@ static int smr_av_insert(struct fid_av *av_fid, const void *addr, size_t count,
 	util_av = container_of(av_fid, struct util_av, av_fid);
 	smr_av = container_of(util_av, struct smr_av, util_av);
 
+	if (smr_av->used + count > SMR_MAX_PEERS) {
+		FI_WARN(&smr_prov, FI_LOG_AV,
+			"AV insert failed. Expect inserting %zu AV entries, but only %zu available\n",
```

Comment:
I don't think this really matters but would it be better to insert as many as possible before the limit is reached and just return the number of addresses that were inserted or fail before? Ie check this on every iteration of the loop and fail in the middle? @shefty?

Reply:
Moving the check into the loop would help make the behavior correct.  This is failing all insertions, but not writing error entries to the EQ for the failed addresses.

Reply:
Hmm. I feel it does not make too much sense, if we know we are going to fail eventually.

Reply:
I would move it to the loop so we can at least report failures to the EQ

Reply:
Like this 
```
for (i = 0; i < count; i++, addr = (char *) addr + strlen(addr) + 1) {
        if (smr_av->used < SMR_MAX_PEERS) {
                ep_name = smr_no_prefix(addr);
                ret = ofi_av_insert_addr(util_av, ep_name, &index);
        } else {
                FI_WARN(&smr_prov, FI_LOG_AV,
                        "AV insert failed. The maximum number of AV "
                        "entries shm supported has been reached.");
                ret = -FI_ENOSYS;
        }
        if (ret) {
                if (util_av->eq)
                        ofi_av_write_event(util_av, i, -ret, context);
       ...
```

Reply:
Basically, but I would use a different error code, such as ENOMEM or ENOSPC

Reply:
Updated.